subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoPlugin.java

```diff
@@ -179,7 +179,7 @@ public void execute(JacocoReport reportTask) {
     }
 
     private void configureJacocoReportDefaults(final JacocoPluginExtension extension, final JacocoReport reportTask) {
-        reportTask.getReports().all(new Action<ConfigurableReport>() {
+        reportTask.getReports().withType(ConfigurableReport.class, new Action<ConfigurableReport>() {
```

Comment:
Right. My point was that we didn't actually need to make a breaking change to the reports container, as people can easily use `withType(ConfigurableReport, Action)` instead. However, if we really do end up making the breaking change, then we may as well just keep using `all(Action)`. What we shouldn't do is both things (breaking change _and_ use `withType`).